mcp-spi/src/main/java/io/modelcontextprotocol/schema/McpSchemaCodec.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ */
+
+package io.modelcontextprotocol.schema;
+
+import java.io.IOException;
+
+/**
+ * Codec interface for encoding and decoding JSON-RPC messages.
+ *
+ * @author Aliaksei Darafeyeu
+ */
+public interface McpSchemaCodec {
```

Comment:
The byte[] base encode and decode as well as `decodeBytes` never seem to be used.  The actual encoded type of the data is likely dependent on the specific transport being used.  I wonder if it might make more sense to change this to something like:

```java
public interface McpSchemaCodec<T> {
  T encode(McpSchema.JSONRPCMessage message) throws IOException;
  McpSchema.JSONRPCMessage decode(T message) throws IOException;
  <U> U unmarshalFrom(Object data, Class<T> clazz);
}
```

Then, each transport would be capable of using the codecs for the encoded data types it understands instead of each codec needing to support all possible encodings.